services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/VxlanFlowSpec.groovy

```diff
@@ -0,0 +1,418 @@
+    Provider<TraffExamService> traffExamProvider
+
+    @Unroll
+    @Tags(HARDWARE)
+    def "System allows to create/update encapsulation type for a flow\
+(#encapsulationCreate.toString() -> #encapsulationUpdate.toString())"() {
+        given: "Two active switches with traffgens"
+        def noviflowSwitchesIds = topology.activeSwitches.findAll { it.noviflow }*.dpId
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeTraffGens*.switchConnected.findAll {
+            noviflowSwitchesIds.contains(it.dpId)
+        } ?: assumeTrue("Should be at least two active traffgens connected to NoviFlow switches for test execution",
+                false)
+
+        when: "Create a flow with #encapsulationCreate.toString() encapsulation type"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.encapsulationType = encapsulationCreate
+        flowHelper.addFlow(flow)
+
+        then: "Flow is created with the #encapsulationCreate.toString() encapsulation type"
+        def flowInfo = northbound.getFlow(flow.id)
+        flowInfo.encapsulationType == encapsulationCreate.toString().toLowerCase()
+
+        //TODO(andriidovhan) check rules(tunnel-id) when pr2503 is merged
+
+        and: "Flow is valid"
+        northbound.validateFlow(flow.id).each { direction -> assert direction.asExpected }
+
+        and: "The flow allows traffic"
+        def traffExam = traffExamProvider.get()
+        def exam = new FlowTrafficExamBuilder(topology, traffExam).buildBidirectionalExam(flow, 0)
+        [exam.forward, exam.reverse].each { direction ->
+            def resources = traffExam.startExam(direction)
+            direction.setResources(resources)
+            assert traffExam.waitExam(direction).hasTraffic()
+        }
+
+        and: "Flow is pingable"
+        def responsePing1 = northbound.pingFlow(flow.id, new PingInput())
+        responsePing1.forward.pingSuccess
+        responsePing1.reverse.pingSuccess
+
+        when: "Try to update the encapsulation type to #encapsulationUpdate.toString()"
+        northbound.updateFlow(flow.id, flow.tap { it.encapsulationType = encapsulationUpdate })
+
+        then: "The encapsulation type is changed to #encapsulationUpdate.toString()"
+        def flowInfo2 = northbound.getFlow(flow.id)
+        flowInfo2.encapsulationType == encapsulationUpdate.toString().toLowerCase()
+
+        and: "Flow is valid"
+        northbound.validateFlow(flow.id).each { direction -> assert direction.asExpected }
+
+        and: "The flow allows traffic"
+        [exam.forward, exam.reverse].each { direction ->
+            def resources = traffExam.startExam(direction)
+            direction.setResources(resources)
+            assert traffExam.waitExam(direction).hasTraffic()
+        }
+
+        and: "Flow is pingable"
+        def responsePing2 = northbound.pingFlow(flow.id, new PingInput())
+        responsePing2.forward.pingSuccess
+        responsePing2.reverse.pingSuccess
```

Comment:
If you run out of variable names, just do this trick:
```
with(northbound.pingFlow(flow.id, new PingInput())) {
  it.forward.pingSuccess
  it.reverse.pingSuccess
}
```
Current implementation is totally fine though